### DISCUSSION TITLE
useFetcher causes an infinite render loop with custom MDX components

### DISCUSSION THREAD

**Discussion by platon-ivanov** [remix-run/remix#8955] [Q&A]
When using [mdx-bundler](https://github.com/kentcdodds/mdx-bundler) or [mdx-js](https://mdxjs.com/guides/mdx-on-demand/) with custom components, they work as expected.

However, if a line with `useFetcher()` is added on any level of the components, it causes the component to rerender infinitely.

**Example:**

Works:
```tsx
<MDXContent
    components={{
      img: props => {
        return <img className="max-w-full" {...props} />
      }}
/>
```

Causes a render loop:
```tsx
<MDXContent
    components={{
      img: props => {
        const fetcher = useFetcher()
        return <img className="max-w-full" {...props} />
      }}
/>
```

How can the infinite render loop be avoided?

**Comment by Pouet--** [comment]
Same problem here, with no external library (just a component with html tags) : declaring useFetcher without using it causes infinite rendering loop.
